kratos/processes/reorder_and_optimize_modelpart_process.cpp

```diff
@@ -231,25 +233,47 @@ namespace Kratos
             
             //reorder
             mrModelPart.Nodes().Sort();
-        }
-        
-
-        
-        
-	std::string ReorderAndOptimizeModelPartProcess::Info() const {
-		return "ReorderAndOptimizeModelPartProcess";
-	}
 
-	void ReorderAndOptimizeModelPartProcess::PrintInfo(std::ostream& rOStream) const {
-		rOStream << Info();
-	}
-
-	void ReorderAndOptimizeModelPartProcess::PrintData(std::ostream& rOStream) const {
+            ReorderElements();
+        }
 
-	}
+        void ReorderAndOptimizeModelPartProcess::ReorderElements()
+        {
+            // Expects element ids are ordered 1 ... Elements().size().
+            std::vector<std::size_t> element_ids(mrModelPart.NumberOfElements());
+            std::vector<std::size_t> node_ids(mrModelPart.NumberOfElements());
+            #pragma omp parallel for
+            for(int i=0; i < static_cast<int>(element_ids.size()); ++i)
+            {
+                auto it = mrModelPart.ElementsBegin() + i;
+                element_ids.at(it->Id() - 1) = it->Id();
+                std::size_t node_id = it->GetGeometry()[0].Id();
+                for (const auto& r_node : it->GetGeometry().Points())
+                    node_id = std::min(node_id, r_node.Id());
+                node_ids.at(it->Id() - 1) = node_id;
```

Comment:
why are u using at() instead of []?

Reply:
To check against array bounds since the function assumes a certain node number. In case the node numbering is changed, it will throw an out_of_range.